libarchive/write.py

```diff
@@ -73,6 +74,39 @@ def add_files(self, *paths):
                         write_finish_entry(write_p)
                         entry_clear(entry_p)
 
+    def add_entry_from_memory(self, entry):
```

Comment:
Please change it. It's inconsistent and confusing, because the type of `entry` is `ArchiveEntry` elsewhere in the code, but here it's a "virtual" `Entry` type that the library doesn't provide (in the test you create it with `namedtuple`). That would probably be okay in JavaScript code, but this is Python.